### DISCUSSION TITLE
useQueries is too slow

### DISCUSSION THREAD

**Discussion by jcready** [TanStack/query#3968] [General]
Even when I attempt to cache at every possible layer, using `useQueries` with hundreds of query options in an array (cached) is about 100x slower than having a single `useQuery` that just runs the same promises in a `Promise.all()`. This is after all those queries have resolved and should no longer change. Just re-rendering my component is dog slow when using `useQueries`.

**Comment by TkDodo** [comment]
can you show a reproduction?

useQueries will re-render the component every time one of the promises resolves, because we cannot know if you want to wait for all promises to resolve before showing something. It could be that data should be shown as early as possible.
